pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -575,7 +575,7 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[api.UniqueVolumeName]*r
 			continue
 		}
 		if volume.pluginIsAttachable {
-			err = rc.actualStateOfWorld.MarkDeviceAsMounted(volume.volumeName)
+			err = rc.actualStateOfWorld.MarkDeviceAsMounted(volume.volumeName, volume.devicePath)
```

Comment:
Here's your problem. This `updateStates()` method is called when a volume is constructed based on state read from disk and its `devicePath` is populated by information fetched from the node (aka attach/detach controller). That means that a reconstructed azure or rbd volume will have the incorrect devicePath (since Mount is not called again).

Really the problem here is that there are two different values "devicePathFromNode" which, because of Azure and RBD may or may not actually be a devicePath, and the "real" `devicePath` which is generated by `WaitForAttach(...)`. Maybe for attachable volumes, `updateStates()` could first call `WaitForAttach(...)` to get the generated devicePath?

+@jingxu97 Thoughts?